### PR TITLE
Address `bbs_database topic-extract` inconsistency unzipping for different sources

### LINKED ISSUE #551: `bbs_database topic-extract` inconsistent behavior for different sources - unzipping
## Description
Current logic

* `source=pubmed`: `bbs_database download` -> `.xml.gz` file -> manual unzipping (`gzip -d  *.xml.gz`) ->  `.xml` file--> `topic-extract`
* `source=biorxiv/medrxiv`: `bbs_database download` -> `.meca` file -> `topic-extract` (unzipping implemented inside) 

## Possible solutions
1. No action. We just need to make sure that once we design the pipeline script we take it into account.
2. Remove unzipping logic from `source=biorxiv/medrxiv`. IMO this one would be a bad idea since after unzipping there are multiple files and one needs to programatically find a unique `.xml` among them.
3. Add unzipping logic to `source=pubmed`. Might make the code + tests a little bit less readable.

### PR DESCRIPTION
**Pull request by EmilieDel**
Fixes #551.

## Description

Please provide here a summary of the changes introduced by this PR.

## How to test?

Please provide here instructions on how to test the changes introduced by this PR.
(if some changes cannot be tested by automated tests)

## Checklist

- [x] This PR refers to an issue from the [issue tracker](https://github.com/BlueBrain/Search/issues).
  (if it is not the case, please create an issue first).
- [x] Unit tests added.
  (if needed)
- [x] Documentation and `whatsnew.rst` updated.
  (if needed)
- [x] `setup.py` and `requirements.txt` updated with new dependencies.
  (if needed)
- [x] Type annotations added.
  (if a function is added or modified)
- [x] All CI tests pass.

### CODE REVIEW

**Review by jankrepl — APPROVED** [review]
Great! Thank you!

File: src/bluesearch/entrypoint/database/topic_extract.py [hunk 1]
```diff
@@ -160,7 +161,10 @@ def run(
     elif article_source is ArticleSource.PUBMED:
         for path in inputs:
             logger.info(f"Processing {path}")
-            articles = ElementTree.parse(input_path)
+            with gzip.open(path) as mygzip:
+                articles_str = mygzip.read()
+            articles = ElementTree.fromstring(articles_str)
+
```
Comment (FrancescoCasalegno) [inline src/bluesearch/entrypoint/database/topic_extract.py:167]: So after this we are assuming in `topic-extract` that the input is a `.xml.gz` file when the source is `PubMed`.
But in `parse` don't we assume we have the `xml` file (i.e. that the `.gz` was unzziped)?
https://github.com/BlueBrain/Search/blob/6e362760ef0d2bbb6cd583dfb363313d222737e2/src/bluesearch/entrypoint/database/parse.py#L130-L136

Maybe this is more a question for @jankrepl who opened #551.

Comment (EmilieDel) [reply src/bluesearch/entrypoint/database/topic_extract.py:167]: I think that as soon as we can do without unzipping and use directly the raw files, it is better (less intermediate/manual) steps. If there are other inconsistencies, we should maybe address them. Do you want me to make the changes in this in the current PR?

Comment (FrancescoCasalegno) [reply src/bluesearch/entrypoint/database/topic_extract.py]: I am fine leaving this part outside of the current PR, but maybe just open an issue with a reference to this PR so we can address it next :)

**Pull request merged**